tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Steps/RegistrationSteps.cs

```diff
@@ -7,19 +10,33 @@ namespace NHSD.BuyingCatalogue.Identity.Api.IntegrationTests.Steps
     [Binding]
     internal sealed class RegistrationSteps
     {
-        private readonly string discovery;
+        private readonly ScenarioContext _context;
+        private readonly string _discovery;
         private readonly SeleniumContext _seleniumContext;
 
-        public RegistrationSteps(IConfiguration configuration, SeleniumContext seleniumContext)
+        public RegistrationSteps(
+            IConfiguration configuration,
+            ScenarioContext context,
+            SeleniumContext seleniumContext)
         {
-            discovery = configuration.GetValue<string>("DiscoveryAddress");
+            _context = context;
+            _discovery = configuration.GetValue<string>("DiscoveryAddress");
             _seleniumContext = seleniumContext;
         }
 
-        [When(@"the user navigates to identity url (.*)")]
+        [When(@"the user navigates to identity url (\S*)$")]
         public void WhenTheUserNavigatesToUrl(string url)
         {
-            _seleniumContext.WebDriver.Navigate().GoToUrl($"{discovery}/{url}");
+            _seleniumContext.WebDriver.Navigate().GoToUrl($"{_discovery}/{url}");
+        }
+
+        [When(@"the user navigates to identity url (\S*) with a valid password reset token")]
+        public void WhenTheUserNavigatesToUrlWithValidPasswordResetToken(string url)
+        {
+            var user = (IdentityUser)_context[ScenarioContextKeys.IdentityUser];
```

Comment:
Just a suggestion but the ScenarioContext has a `Get<T>` method that can retrieve a typed entry. I find it easier to read be clearly subjective, so don't feel you have to make the change.

Reply:
Thanks for point that out – it's much nicer.